pkg/sources/job_progress.go

```diff
@@ -8,6 +8,20 @@ import (
 	"fmt"
 	"sync"
 	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/common"
+)
+
+var (
+	hooksExecTime = promauto.NewHistogramVec(prometheus.HistogramOpts{
+		Namespace: common.MetricsNamespace,
+		Subsystem: common.MetricsSubsystem,
+		Name:      "hooks_exec_time_ms",
+		Help:      "Time spent executing hooks (ms)",
+		Buckets:   []float64{5, 10, 25, 50, 100, 250, 500, 1000, 2500, 5000},
```

Comment:
question: have you tested this in a dev/local env? 

Since this needs to be sent over the network even with compression there is a chance this exceeds our current gRPC payload limit. I only mention this because this one of the main reasons I had to change our metrics for detectors to only include 3-4 bucket values due to the enormous size of the payload.

Reply:
I added compression and also increased the payload limit on the API server, but i'm not sure if that would be enough for these metrics.

Reply:
Ah, good callout. I wasn't aware that the number of buckets affected the size of the payload. These buckets were copied from another metric in thog (but on the API side, so no GRPC messaging). I'll reduce them.

Reply:
Funnily enough I did the exact same thing and copied the exact same metrics when I added them for our detectors 🤣 .

Reply:
Ooh, I didn't know that either.